src/internal.c

```diff
@@ -6697,11 +6702,441 @@ int SendServiceAccept(WOLFSSH* ssh, byte serviceId)
+    int ret = WS_SUCCESS;
+
+    if (ssh == NULL || output == NULL || idx == NULL || authData == NULL ||
+            sigStart == NULL || keySig == NULL)
+        ret = WS_BAD_ARGUMENT;
+
+    begin = *idx;
+
+    if (ret == WS_SUCCESS) {
+        hashId = HashForId(keySig->keySigId);
+        WMEMSET(digest, 0, sizeof(digest));
+        digestSz = wc_HashGetDigestSize(hashId);
+        ret = wc_HashInit(&hash, hashId);
+    }
+
+    if (ret == WS_SUCCESS) {
+        c32toa(ssh->sessionIdSz, digest);
+        ret = wc_HashUpdate(&hash, hashId, digest, UINT32_SZ);
+    }
+
+    if (ret == WS_SUCCESS)
+        ret = wc_HashUpdate(&hash, hashId,
+                ssh->sessionId, ssh->sessionIdSz);
+
+    if (ret == WS_SUCCESS)
+        ret = wc_HashUpdate(&hash, hashId,
+                 sigStart, begin - sigStartIdx);
+
+    if (ret == WS_SUCCESS)
+        ret = wc_HashFinal(&hash, hashId, digest);
+
+    if (ret == WS_SUCCESS) {
+        byte encDigest[MAX_ENCODED_SIG_SZ];
+        int encDigestSz;
+
+        c32toa(keySig->sigSz + 7 + LENGTH_SZ * 2, output + begin);
+        begin += LENGTH_SZ;
+        c32toa(7, output + begin);
+        begin += LENGTH_SZ;
+        WMEMCPY(output + begin, "ssh-rsa", 7);
+        begin += 7;
+        c32toa(keySig->sigSz, output + begin);
+        begin += LENGTH_SZ;
+        encDigestSz = wc_EncodeSignature(encDigest, digest, digestSz,
+                wc_HashGetOID(hashId));
+        if (encDigestSz <= 0) {
+            WLOG(WS_LOG_DEBUG, "SUAR: Bad Encode Sig");
+            ret = WS_CRYPTO_FAILED;
+        }
+        else {
+            int sigSz;
+            WLOG(WS_LOG_INFO, "Signing hash with RSA.");
+            sigSz = wc_RsaSSL_Sign(encDigest, encDigestSz,
+                    output + begin, keySig->sigSz,
+                    &keySig->ks.rsa.key, ssh->rng);
+            if (sigSz <= 0 || (word32)sigSz != keySig->sigSz) {
+                WLOG(WS_LOG_DEBUG, "SUAR: Bad RSA Sign");
+                ret = WS_RSA_E;
+            }
+        }
+
+        if (ret == WS_SUCCESS) {
+            begin += keySig->sigSz;
+            *idx = begin;
+        }
+    }
+
+    return ret;
+}
+
+
+static int PrepareUserAuthRequestEcc(WOLFSSH* ssh, word32* payloadSz,
+        const WS_UserAuthData* authData, WS_KeySignature* keySig)
+{
+    int ret = WS_SUCCESS;
+
+    WLOG(WS_LOG_DEBUG, "Entering PrepareUserAuthRequestEcc()");
+    if (ssh == NULL || payloadSz == NULL || authData == NULL || keySig == NULL)
+        ret = WS_BAD_ARGUMENT;
+
+    if (ret == WS_SUCCESS)
+        ret = wc_ecc_init(&keySig->ks.ecc.key);
+
+    if (ret == WS_SUCCESS) {
+        word32 idx = 0;
+        ret = wc_EccPrivateKeyDecode(authData->sf.publicKey.privateKey,
+                &idx, &keySig->ks.ecc.key,
+                authData->sf.publicKey.privateKeySz);
+    }
+
+    if (ret == WS_SUCCESS) {
+        if (authData->sf.publicKey.hasSignature) {
+            int sigSz = wc_ecc_sig_size(&keySig->ks.ecc.key);
+            sigSz = 64;
```

Comment:
Should sigSz be forced to 64 here, when the line above it sets it based on wc_ecc_sig_size()?

Reply:
Oh yeah, that was for something I was testing and forgot to remove. Thanks for catching that.